### PR TITLE
Fix big int and base value issue

### PR DESCRIPTION
**Pull request by Ehsan-saradar**
Fix:

telegram-cloud-photo-size-2-5190640691195874380-y

## Summary by CodeRabbit

* **New Features**
  * Added test tooling and package scripts for running tests and a test UI.

* **Bug Fixes**
  * Billing and transaction amounts are now truncated before integer conversion to prevent precision/display issues.

* **Tests**
  * Added comprehensive pricing-formatting tests covering one-time, per-transaction, recurring (monthly/yearly/weekly), edge cases, multi-currency, and regressions.

### CODE REVIEW

File: src/utils/functions.test.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        baseValue,
+        currency,
+        frequency: "",
+        type: "per-tx",
+      });
+
+      expect(result).toContain("per transaction");
+      // Should not throw BigInt error
+    });
+  });
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply src/utils/functions.test.ts]: Please check again
